src/Common/Core/Impl/Extensions/TaskExtensions.cs

```diff
@@ -56,14 +58,41 @@ public static class TaskExtensions {
         }
 
         /// <summary>
-        /// Suppresses warnings about unawaited tasks and ensures that unhandled
-        /// errors will cause the process to terminate.
+        /// Rethrows task exceptions back to the callers synchronization context if it is possible
         /// </summary>
         /// <remarks>
         /// <see cref="OperationCanceledException"/> is always ignored.
         /// </remarks>
-        public static async void DoNotWait(this Task task) {
-            await task.SilenceException<OperationCanceledException>();
+        public static void DoNotWait(this Task task) {
```

Comment:
I think the point here is to make it so that exceptions get propagated to the main event loop in case UI thread calls `DoNotWait` (which is by far the most common use), so that any top-level error handling/reporting/logging facilities can kick in.